R/pkg/tests/fulltests/test_sparkSQL.R

```diff
@@ -2684,21 +2684,21 @@ test_that("union(), unionByName(), rbind(), except(), and intersect() on a DataF
   writeLines(lines, jsonPath2)
   df2 <- read.df(jsonPath2, "json")
 
-  unioned <- arrange(union(df, df2), df$age)
+  unioned <- arrange(union(df, df2), "age")
```

Comment:
NOTE: Since `Union` assigns new `exprID`s now in output, we cannot use `df$age` here.

Reply:
ah so this is a breaking change. How many tests like this are broken by this PR?

Reply:
If there are many, we may need to skip `Union` in the exprId check as a workaround.

Reply:
Ah I see. Only these tests in this file failed in  the existing end-2-end tests, but Scala/Python DataFrame has the same behaivour change, so I'll revert the latest commit and ignore the attributes of `Union` in integrity checks.

Reply:
Updated: https://github.com/apache/spark/pull/29585/files#diff-27c76f96a7b2733ecfd6f46a1716e153R232-R238